src/metabase/lib/underlying.cljc

```diff
@@ -0,0 +1,67 @@
+(ns metabase.lib.underlying
```

Comment:
It doesn't really have anything to do with drill-thru. There are other callers of `topLevelQuery` in the FE.

(If `drill-filter` seems out of place here, it could go somewhere under `metabase.lib.drill-thru.*`.)

Reply:
Yes, I was confused by the mix.

Reply:
I've moved `drill-filter` as a private function `metabase.lib.drill-thru.underlying-records/drill-filter`.